packages/loopring_v3/contracts/core/impl/ExchangeV3.sol

```diff
@@ -613,6 +619,23 @@ contract ExchangeV3 is IExchangeV3
         emit TransactionApproved(owner, transactionHash);
     }
 
+    function approveTransactions(
+        address[] calldata owners,
+        bytes32[] calldata transactionHashes
+        )
+        external
+        override
+        nonReentrant
+        onlyFromUserOrAgent(owner)
+    {
+        require(owners.length == transactionHashes.length, "INVALID_DATA");
+        for (uint i = 0; i < owners.length; i++) {
+            require(isUserOrAgent(owners[i]), "UNAUTHORIZED");
```

Comment:
Possible easy optimization here to add the ability to `IAgentRegistry` to batch check an agent against multiple owners:
```
function isAgent(
    address[] calldata users,
    address agent
    )
    external
    override
    view
    returns (bool)
{
    if (isUniversalAgent(agent)) {
        return true;
    }
    for (uint i = 0; i < users.length; i++) {
        if (!isUserAgent(users[i], agent)) {
            return false;
        }
    }
    return true;
}
```

The AMM pool will be a universal agent so instead of a CALL + SREAD / transaction, with this function we have a fixed cost of a single CALL + SREAD.